test/extensions/filters/network/common/fuzz/uber_per_writefilter.cc

```diff
@@ -30,6 +32,26 @@ std::vector<absl::string_view> UberWriteFilterFuzzer::filterNames() {
   return filter_names;
 }
 
+void UberWriteFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name,
+                                                       Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some input that crashes the
+  // mock/fake objects are also prohibited. We could also avoid fuzzing some unfinished features by
+  // checking them here. For now there is only one filter {MongoProxy} on which we have a
+  // constraint.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().MongoProxy) {
+    envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy& config =
+        dynamic_cast<envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy&>(
+            *config_message);
+    if (config.has_delay() && config.mutable_delay()->has_header_delay()) {
+      // Mongo has no header map. Only fixed_delay is supported.
+      throw EnvoyException(
```

Comment:
Explain what a "header map" is in this context.  Why does the lack of a header map mean that only fixed delay is supported?

Reply:
> Explain what a "header map" is in this context. Why does the lack of a header map mean that only fixed delay is supported?

Added! 
    ```
  // MongoProxy filter doesn't allow header_delay because it will pass nullptr to percentage()
      // which will cause "runtime error: member call on null pointer". (See:
      // https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/network/mongo_proxy/proxy.cc#L403
      // and
      // https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/common/fault/fault_config.cc#L16)
```